### ISSUE TITLE
HuggingFace Hub Models with Simple API (AutoModelForCausalLM + .score() method)

### ISSUE THREAD

**Issue by mapo80** [Junjie-Gao19/DeQA-Doc#3]
Hi! First of all, congratulations on winning the VQualA 2025 DIQA Challenge! 🏆

I've uploaded your DeQA-Doc models to HuggingFace Hub with a simplified API. The main goals were:

1. **Simplify usage**: Make the models as easy to use as possible, similar to [DeQA-Score](https://github.com/zhiyuanyou/DeQA-Score), with a single `.score()` method that handles all preprocessing internally
2. **Test functionality**: Verify that the models work correctly when loaded directly from HuggingFace Hub with `trust_remote_code=True`

## Models Published

| Model | HuggingFace Hub |
|-------|-----------------|
| Overall | [mapo80/DeQA-Doc-Overall](https://huggingface.co/mapo80/DeQA-Doc-Overall) |
| Color | [mapo80/DeQA-Doc-Color](https://huggingface.co/mapo80/DeQA-Doc-Color) |
| Sharpness | [mapo80/DeQA-Doc-Sharpness](https://huggingface.co/mapo80/DeQA-Doc-Sharpness) |

## Simple Usage

```python
import torch
from transformers import AutoModelForCausalLM
from PIL import Image

# Load model directly from HuggingFace
model = AutoModelForCausalLM.from_pretrained(
    "mapo80/DeQA-Doc-Overall",  # or Color, Sharpness
    trust_remote_code=True,
    torch_dtype=torch.float16,
    device_map="auto",
)

# Score an image with a single method call
image = Image.open("document.jpg").convert("RGB")
score = model.score([image])
print(f"Quality Score: {score.item():.2f} / 5.0")

# Batch processing is also supported
scores = model.score([image1, image2, image3])
```

## What Was Done

1. Added custom model files for HuggingFace `trust_remote_code` compatibility:
   - `modeling_mplug_owl2_huggingface.py` with `.score()` method
   - `configuration_mplug_owl2.py`
   - `config.json` with `auto_map` for automatic class loading

2. The `.score()` method handles all preprocessing internally:
   - Image loading and resizing
   - Tokenization
   - Inference
   - Score extraction from logits

3. Full attribution to your work in all README files with proper citation

## Hardware Requirements

- ~16 GB VRAM (fp16) per model
- Tested on NVIDIA A40 (46GB)
- Works with `device_map="auto"` for CPU offload on smaller GPUs

---

Let me know if you'd like any changes or if you'd prefer to host these models under your own HuggingFace organization. Happy to transfer them or help in any way!

Thanks for the great work on DeQA-Doc!
